hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/UnmodifiableConfiguration.java

```diff
@@ -0,0 +1,545 @@
+  }
+
+  @Override
+  public void setAllowNullValueProperties(boolean val) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setRestrictSystemProps(boolean val) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void set(String name, String value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void set(String name, String value, String source) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public synchronized void unset(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public synchronized void setIfUnset(String name, String value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setInt(String name, int value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setLong(String name, long value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setFloat(String name, float value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setDouble(String name, double value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setBoolean(String name, boolean value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setBooleanIfUnset(String name, boolean value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <T extends Enum<T>> void setEnum(String name, T value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setTimeDuration(String name, long value, TimeUnit unit) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setStorageSize(String name, double value, StorageUnit unit) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setPattern(String name, Pattern pattern) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setStrings(String name, String... values) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setSocketAddr(String name, InetSocketAddress addr) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public InetSocketAddress updateConnectAddr(String hostProperty, String addressProperty,
+      String defaultAddressValue, InetSocketAddress addr) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public InetSocketAddress updateConnectAddr(String name, InetSocketAddress addr) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void setClass(String name, Class<?> theClass, Class<?> xface) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void clear() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void writeXml(OutputStream out) {
```

Comment:
These are probably okay to proxy.

Reply:
By that, I mean that is probably okay to provide an implementation that calls `this.other.writeXml(...)`